src/gval/accessors/gval_xarray.py

```diff
@@ -113,9 +114,13 @@ def categorical_compare(
 
         self.check_same_type(benchmark_map)
 
+        candidate, benchmark = _align_numeric_data_type(
```

Comment:
I'm curious if both `_align_numeric_data_type()` and `_spatial_alignment()` return copies of the candidates or references to the original objects? If they return copies could these be leading to memory overuse?

Reply:
Changed to be by reference only